projects/packages/search/src/class-stats.php

```diff
@@ -34,4 +38,44 @@ public function get_stats_from_wpcom() {
 
 		return $response;
 	}
+
+	/**
+	 * Estimate record counts via a local database query.
+	 */
+	public static function estimate_count() {
+		$cached_value = wp_cache_get( self::COUNT_ESTIMATE_CACHE_KEY, self::CACHE_GROUP );
+		if ( false !== $cached_value ) {
+			return $cached_value;
+		}
+
+		global $wpdb;
+		$option_count = (int) get_option( 'post_count', 0 );
+
+		// Excluded post types below are not exhaustive, but enable us to generate a sufficiently accurate count estimate.
+		$db_count = (int) $wpdb->get_var(
```

Comment:
I think this will include the feedback post type (contact form)

I think it may work to get a list of exclusions with:

```
get_post_types( [ 'public' => false, 'exclude_from_search' => true ], 'names', 'or' )
```

On my test wpcom site this gives me:
```
array (
  'revision' => 'revision',
  'nav_menu_item' => 'nav_menu_item',
  'custom_css' => 'custom_css',
  'customize_changeset' => 'customize_changeset',
  'oembed_cache' => 'oembed_cache',
  'user_request' => 'user_request',
  'wp_block' => 'wp_block',
  'wp_template' => 'wp_template',
  'wp_template_part' => 'wp_template_part',
  'wp_global_styles' => 'wp_global_styles',
  'wp_navigation' => 'wp_navigation',
  'amp_validated_url' => 'amp_validated_url',
  'feedback' => 'feedback',
  'attachment_proxy' => 'attachment_proxy',
  'safecss' => 'safecss',
  'custom_dns' => 'custom_dns',
  'dns_revision' => 'dns_revision',
  'jp_mem_plan' => 'jp_mem_plan',
  'jp_pay_order' => 'jp_pay_order',
  'jp_pay_product' => 'jp_pay_product',
)
```

Or a list of inclusions with:

```
get_post_types( [ 'public' => true, 'exclude_from_search' => false ], 'names', 'and' )
```

Test site gives me:
```
array (
  'post' => 'post',
  'page' => 'page',
  'attachment' => 'attachment',
)
```

Merging one of the above with a hardcoded list is probably necessary though because things like bbpress are a little special and there will likely be other cases like that.

Probably could also get the list of post_status with:
```
get_post_stati( array( 'public' => true ) )
```

The other concern I have here (which we can't do too much about) is that depending on when this code runs all post types may not have yet been registered which is another reason to have it at least partly based on a hard coded list.

Reply:
I implemented your suggested approach to creating a post types exclusion list by combining a hardcoded list with one generated by `get_post_types`. Thanks! 👍